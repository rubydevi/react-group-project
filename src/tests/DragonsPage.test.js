import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import DragonsPage from '../components/Dragons';

const mockStore = configureMockStore([]);

// Mock the axios module
jest.mock('axios', () => ({
  get: jest.fn(() => Promise.resolve({ data: [] })),
}));

describe('DragonsPage Component', () => {
  let store;
  let component;

  beforeEach(() => {
    // Mock dragons data
    const initialState = {
      dragons: {
        dragons: [
          {
            id: 1,
            name: 'Dragon 1',
            description: 'Description for Dragon 1',
            flickr_images: ['url1', 'url2'],
            reserved: false,
          },
          {
            id: 2,
            name: 'Dragon 2',
            description: 'Description for Dragon 2',
            flickr_images: ['url3', 'url4'],
            reserved: true,
          },
        ],
      },
    };

    store = mockStore(initialState);
    store.dispatch = jest.fn();

    component = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <DragonsPage />
        </MemoryRouter>
      </Provider>,
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
