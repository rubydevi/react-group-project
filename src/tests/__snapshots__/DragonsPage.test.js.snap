// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragonsPage Component renders correctly 1`] = `
<section
  className="dragons"
>
  <div
    className="dragon"
  >
    <div
      className="dragonImg"
    >
      <img
        alt="Dragon 1"
        src="url1"
      />
    </div>
    <div
      className="dragonInfo"
    >
      <h2>
        Dragon 1
      </h2>
      <p>
        Description for Dragon 1
      </p>
      <button
        className="reserve"
        onClick={[Function]}
        type="button"
      >
        Reserve Dragon
      </button>
    </div>
  </div>
  <div
    className="dragon"
  >
    <div
      className="dragonImg"
    >
      <img
        alt="Dragon 2"
        src="url3"
      />
    </div>
    <div
      className="dragonInfo"
    >
      <h2>
        Dragon 2
      </h2>
      <p>
        <span
          className="reservedText"
        >
          Reserved
        </span>
        Description for Dragon 2
      </p>
      <button
        className="cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</section>
`;
